ethdb/privateapi/ethbackend.go

```diff
@@ -335,6 +356,87 @@ func (s *EthBackendServer) EngineForkChoiceUpdatedV1(ctx context.Context, req *r
 	}, nil
 }
 
+func (s *EthBackendServer) StartProposer() {
+	go func() {
+		atomic.StoreUint32(s.pauseAssemble, 1)
+		for {
+			if atomic.LoadUint32(s.pauseAssemble) == 1 || atomic.LoadUint32(s.waitingForBeaconChain) == 0 {
```

Comment:
It's better to use [sync.Cond](https://pkg.go.dev/sync#Cond) for waiting for `pauseAssemble` rather than use a timer. You can then make `pauseAssemble` a bool and protect it by the lock. And no need to check `waitingForBeaconChain` ­- it's done in `EngineForkChoiceUpdatedV1` already.

Reply:
we can actually remove pauseAssemble and just use a `Sync.Cond` i believe